### PR TITLE
[ENH] Classification `_fit_predict` mockup

### PR DESCRIPTION
**Pull request by MatthewMiddlehurst**
For comparison with #1104
See #1102 

Adds a `_fit_predict` and `_fit_predict_proba` method to the classifier base class. This is intended to replace the `_get_train_probs` used currently in some classifiers.

### CODE REVIEW

**Comment by TonyBagnall** [comment]
could you show how get_predict_proba works with this? Could you show with STC? How do you control if fit_predict finds train probas or not?

File: aeon/classification/base.py [hunk 1]
```diff
@@ -354,6 +424,23 @@ def _predict_proba(self, X) -> np.ndarray:
...
+        return self._fit_predict_default(X, y, "predict")
+
+    def _fit_predict_proba(self, X, y) -> np.ndarray:
+        return self._fit_predict_default(X, y, "predict_proba")
+
+    def _fit_setup(self, X, y):
+        # reset estimator at the start of fit
+        self.reset()
+
+        # All of this can move up to BaseCollection
```
Comment (TonyBagnall) [inline aeon/classification/base.py:437]: it cant all be moved up to BaseCollection, check_y is specific for classification, and transformers do not have predict

**Review by TonyBagnall — APPROVED** [review]
I'm fine with doing it this way, and refactor _get_train_probs, removing the check_X and check_y calls.

File: aeon/classification/base.py (hunk 1 above)
Comment (MatthewMiddlehurst) [reply aeon/classification/base.py:437]: Included in #1116

**Comment by MatthewMiddlehurst** [comment]
Draft PR is here #1116.

**Pull request merged**